compose-lint-checks/src/main/java/slack/lint/compose/SlotReusedDetector.kt

```diff
@@ -46,37 +46,21 @@ class SlotReusedDetector : ComposableFunctionDetector(), SourceCodeScanner {
     val slotParameters = method.slotParameters(context.evaluator)
 
     val callExpressions = composableBlockExpression.findChildrenByClass<KtCallExpression>().toList()
-    val innerLambdas = composableBlockExpression.findChildrenByClass<KtLambdaExpression>().toList()
 
-    slotParameters.forEach { slotParameter ->
-      // Find lambdas that shadow this parameter name, to make sure that they aren't shadowing
-      // the references we are looking through
-      val lambdasWithMatchingParameterName =
-        innerLambdas.filter { innerLambda ->
-          // Otherwise look to see if any of the parameters on the inner lambda have the
-          // same name
-          innerLambda.valueParameters
-            // Ignore parameters with a destructuring declaration instead of a named
-            // parameter
-            .filter { it.destructuringDeclaration == null }
-            .any { it.name == slotParameter.name }
-        }
+    val psiManager = PsiManager.getInstance(context.project.ideaProject)
```

Comment:
TIL. Is this safe for us to use though?

Reply:
Good question... I got to this in a chain of warnings/docs.

I initially used `==`, which triggered a [`LintImplPsiEquals`](https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/LintDetectorDetector.kt;l=1538;drc=de1ca7e98fc66a287164ca9b21d65012b0a80e70) lint error and recommends using `isEquivalentTo`.

Then looking at `isEquivalentTo`, the [documentation](https://github.com/JetBrains/intellij-community/blob/213249ff9b4378985ab68391ab186596e604e247/platform/core-api/src/com/intellij/psi/PsiElement.java#L609-L617) says to not call it directly, and instead to use `PsiManager.areElementsEquivalent` instead.

Reply:
How about using `static` util: https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/codeInsight/PsiEquivalenceUtil.java#L97 (which is also used by some places in Kotlin compiler internals)

Reply:
Thank you for the recommendation! Updated to use `PsiEquivalenceUtil` in the latest commit.